src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -267,25 +267,56 @@ private static ZonedDateTime fixYearIfInFuture(ZonedDateTime dateAndTime) {
   }
 
   /**
-   * Splits a string into a stream of strings, and filters out any that don't have at least one
-   * digit.
+   * Splits a string on whitespace and returns a stream of the resulting strings.
    */
   private static Stream<String> getTokensFromRawText(String rawText) {
-    return Stream.of(rawText.split("\\s")).filter(ReceiptAnalysis::hasDigit);
+    return Stream.of(rawText.split("\\s"));
   }
 
   /**
-   * Checks if the token contains at least one digit.
+   * Checks if the token is formatted as a date.
    */
-  private static boolean hasDigit(String token) {
-    return digitRegex.matcher(token).matches();
+  private static boolean isDate(String token) {
+    return dateRegex.matcher(token).matches();
   }
 
   /**
-   * Checks if the token is formatted as a date.
+   * Checks the raw text in the builder for prices that can be parsed. The largest price found, if
+   * it exists, is added to the builder.
    */
-  private static boolean isDate(String token) {
-    return dateRegex.matcher(token).matches();
+  private static void checkForParsablePrices(AnalysisResults.Builder analysisBuilder) {
+    // Assume that the largest price on the receipt is the total price.
+    double largestPrice = getTokensFromRawText(analysisBuilder.getRawText())
+                              .filter(ReceiptAnalysis::isPrice)
+                              .mapToDouble(ReceiptAnalysis::parsePrice)
+                              .reduce(Double.NEGATIVE_INFINITY, Double::max);
```

Comment:
Instead of using Double.NEGATIVE_INFINITY to represent a price that can't be parsed, I would use Optional. You can have parsePrice return Optional<Double>, and then filter out empty optionals.

Reply:
Again, I avoided Optionals because I preferred the simpler method rather than adding in an extra filter and map. In this case, I also chose to use an identity instead of Optionals because that's how I learned to use reduce in ML, so it seemed more natural.